Source/Core/AudioCommon/XAudio2_7Stream.cpp

```diff
@@ -121,7 +160,14 @@ void StreamingVoiceContext2_7::OnBufferEnd(void* context)
   // m_sound_sync_event->Wait(); // sync
   // m_sound_sync_event->Spin(); // or tight sync
 
-  m_mixer->Mix(static_cast<short*>(context), SAMPLES_PER_BUFFER);
+  if (m_use_surround)
+  {
+    m_mixer->MixSurround(static_cast<float*>(context), SAMPLES_PER_BUFFER_SURROUND);
```

Comment:
This cast, and the other one below it are technically undefined behavior, since it's upcasting a `u8` buffer, which breaks alignment requirements.

Reply:
Understood. I guess I could change it from static_cast to reinterpret_cast so it can show what it is really doing? I suppose that won't correct the alignment breakage, right?

Reply:
> I suppose that won't correct the alignment breakage, right?

It wouldn't. You'd need to use a `std::variant` such as:

```cpp
std::variant<std::unique_ptr<short[]>, std::unique_ptr<float[]>> xaudio_buffer;
```

which would preserve the type of buffer. As opposed to just being a `BYTE` buffer (these are allowed to be `reinterpret_cast`ed to `BYTE` (aka `unsigned char`) without breaking alignment requirements.

Reply:
Is it correct now, or should I change anything else?

Reply:
Yup mostly good. Just cosmetic stuff: the `reinterpret_cast`s in `Mix` and `MixSurround` calls can remain `static_casts`. However the bodies of statements that travel more than one line should be braced.

e.g.

```cpp
for (int i = 0; i != NUM_BUFFERS; ++i)
  SubmitBuffer(
      reinterpret_cast<BYTE*>(std::get<std::unique_ptr<float[]>>(xaudio_buffer).get()) +
      (i * BUFFER_SIZE_BYTES_SURROUND));
```

should be:

```cpp
for (int i = 0; i != NUM_BUFFERS; ++i)
{
  SubmitBuffer(
      reinterpret_cast<BYTE*>(std::get<std::unique_ptr<float[]>>(xaudio_buffer).get()) +
      (i * BUFFER_SIZE_BYTES_SURROUND));
}
```